src/views/pacbio/PacbioRunShow.vue

```diff
@@ -68,79 +72,44 @@ export default {
       type: [String, Number],
       default: 0,
     },
-    actions: {
-      type: Object,
-      default() {
-        return {
-          create: {
-            id: 'create',
-            theme: 'create',
-            label: 'Create',
-            method: 'createRun',
-          },
-          update: {
-            id: 'update',
-            theme: 'update',
-            label: 'Update',
-            method: 'updateRun',
-          },
-        }
-      },
-    },
-  },
-  data() {
-    return {
-      newRecord: isNaN(this.id),
-    }
   },
   computed: {
-    currentAction() {
-      return this.actions[this.newRecord ? 'create' : 'update']
+    newRecord() {
+      return this.runType.type === RunTypeEnum.New
     },
-    ...mapGetters(['currentRun', 'poolBarcodes']),
-    ...mapState({
-      currentRun: (state) => state.currentRun,
-    }),
+    ...mapGetters(['runType']),
   },
   created() {
```

Comment:
I would move this functionality into a method under the methods section. (e.g methods: { fetcher() { ... } }) . Then add the DataFetcher wrapper around the component and pass it in the fetcher method